batch/batch/aioclient.py

```diff
@@ -228,10 +235,11 @@ def __init__(self, session, url=None, token_file=None, token=None, headers=None)
                       attributes=b.get('attributes'))
                 for b in batches]
 
-    async def get_job(self, id):
-        j = await self._get('/jobs/{}'.format(id))
-        return Job(self,
-                   j['id'],
+    async def get_job(self, batch_id, job_id):
+        b = await self.get_batch(batch_id)
```

Comment:
This opens you up to race conditions. You should grab everything in one query. If batch is mutated between the time that you look up the entry in Batch table and Jobs table, you have corrupted state.

Even if this doesn't seem to be a problem now (say because you don't expect the batch to change), it seems better to avoid that possibility.